### PR TITLE
Clean the `create_array` in IPC reader.

### PR DESCRIPTION
**Pull request by HaoYang670**
# Which issue does this PR close?

None.

# What changes are included in this PR?
Did some code cleaning work when I learned the IPC reader code.

# Are there any user-facing changes?
No.

### CODE REVIEW

File: arrow/src/ipc/reader.rs [hunk 1]
```diff
@@ -433,28 +395,26 @@ fn skip_field(
...
         Utf8 | Binary | LargeBinary | LargeUtf8 => {
-            // read 3 buffers
+            // read 3 buffers: null buffer (optional), offsets buffer and data buffer
             ArrayData::builder(data_type.clone())
                 .len(length)
                 .buffers(buffers[1..3].to_vec())
-                .offset(0)
-                .null_bit_buffer((null_count > 0).then(|| buffers[0].clone()))
+                .null_bit_buffer(null_buffer)
                 .build()
```
Comment (HaoYang670) [inline arrow/src/ipc/reader.rs:409]: Why do we only do validation when building these 4 types array, but for other types, we use `build_uncheked` ?

Comment (tustvold) [reply arrow/src/ipc/reader.rs:409]: IMO we should probably be validating all of them, I'm not sure why we aren't validating some of them... IPC is inherently not trusted

**Review by tustvold — APPROVED** [review]

File: arrow/src/ipc/reader.rs (hunk 1 above)
Comment (HaoYang670) [reply arrow/src/ipc/reader.rs:409]: Will file an issue to track it.

**Comment by liukun4515** [comment]
why the data in the ipc can't be trusted from this issue https://github.com/apache/arrow-rs/issues/2541 ?

**Pull request merged**